model_analyzer/result/result_manager.py

```diff
@@ -318,13 +364,13 @@ def _add_results_to_heaps(self):
         results = self._state_manager.get_state_variable(
             'ResultManager.results')
 
-        for model_name in self._analysis_model_names:
-            model_measurements = results.get_model_measurements_dict(model_name)
+        for model_name in results.get_list_of_models():
```

Comment:
is this correct for multi-model?

Also, the only difference here is that we no longer assert? Is that assert handled elsewhere now for the case it matters?

Reply:
Yes, I believe so. The Results class already has the correct multi-model name (comes from the checkpoint).

There's no assertion here. This is new logic to ensure that we don't add checkpointed models to the heap that the user didn't specify on a rerun. Note: they are still present in the Result class, so these other model results won't be lost when the new checkpoint is created.

Reply:
re-add assert as discussed

Reply:
Assert added. I've had to fix a bunch of unit tests as a fallout from this. Please review the latest change.s